rclcpp/include/rclcpp/node_handle.hpp

```diff
@@ -0,0 +1,280 @@
+        node->get_node_timers_interface() : nullptr;
+      topics_ = (std::is_same_v<TopicsInterface, InterfaceTs>|| ...) ?
+        node->get_node_topics_interface() : nullptr;
+      services_ = (std::is_same_v<ServicesInterface, InterfaceTs>|| ...) ?
+        node->get_node_services_interface() : nullptr;
+      waitables_ = (std::is_same_v<WaitablesInterface, InterfaceTs>|| ...) ?
+        node->get_node_waitables_interface() : nullptr;
+      parameters_ = (std::is_same_v<ParametersInterface, InterfaceTs>|| ...) ?
+        node->get_node_parameters_interface() : nullptr;
+      time_source_ = (std::is_same_v<TimeSourceInterface, InterfaceTs>|| ...) ?
+        node->get_node_time_source_interface() : nullptr;
+    }
+  }
+
+  // Getters and setters
+
+  /// Return the bound NodeBaseInterface
+  inline BaseInterface base() {return base_;}
+
+  /// Set the bound NodeBaseInterface
+  inline void base(BaseInterface interface) {base_ = interface;}
+
+  /// Return the bound NodeClockInterface
+  inline ClockInterface clock() {return clock_;}
+
+  /// Set the bound NodeClockInterface
+  inline void clock(ClockInterface interface) {clock_ = interface;}
+
+  /// Return the bound NodeGraphInterface
+  inline GraphInterface graph() {return graph_;}
+
+  /// Set the bound NodeGraphInterface
+  inline void graph(GraphInterface interface) {graph_ = interface;}
+
+  /// Return the bound NodeLoggingInterface
+  inline LoggingInterface logging() {return logging_;}
+
+  /// Set the bound NodeLoggingInterface
+  inline void logging(LoggingInterface interface) {logging_ = interface;}
+
+  /// Return the bound NodeTimersInterface
+  inline TimersInterface timers() {return timers_;}
+
+  /// Set the bound NodeTimersInterface
+  inline void timers(TimersInterface interface) {timers_ = interface;}
+
+  /// Return the bound NodeTopicsInterface
+  inline TopicsInterface topics() {return topics_;}
+
+  /// Set the bound NodeTopicsInterface
+  inline void topics(TopicsInterface interface) {topics_ = interface;}
+
+  /// Return the bound NodeServicesInterface
+  inline ServicesInterface services() {return services_;}
+
+  /// Set the bound NodeServicesInterface
+  inline void services(ServicesInterface interface) {services_ = interface;}
+
+  /// Return the bound NodeWaitablesInterface
+  inline WaitablesInterface waitables() {return waitables_;}
+
+  /// Set the bound NodeWaitablesInterface
+  inline void waitables(WaitablesInterface interface) {waitables_ = interface;}
+
+  /// Return the bound NodeParametersInterface
+  inline ParametersInterface parameters() {return parameters_;}
+
+  /// Set the bound NodeParametersInterface
+  inline void parameters(ParametersInterface interface) {parameters_ = interface;}
+
+  /// Return the bound NodeTimeSourceInterface
+  inline TimeSourceInterface time_source() {return time_source_;}
+
+  /// Set the bound NodeTimeSourceInterface
+  inline void time_source(TimeSourceInterface interface) {time_source_ = interface;}
```

Comment:
I don't see the point of the setters. Why do we need them?

Reply:
It's to allow building of aggregate node handles (from parts), as suggested [from the discussion in the original issue](https://github.com/ros2/rclcpp/issues/831#issuecomment-530157701).

Reply:
I think those should be done via constructor arguments. You don't need to have setters to do this. I would still remove these.